### ISSUE TITLE
Block direct access to script sources

### ISSUE THREAD

**Issue by t2ym** [t2ym/thin-hook#237]
Block direct access to script sources

## Notes
- `event.request.destination === "document"` for `view-source:*`
  - [Fetch Standard:Request Destination](https://fetch.spec.whatwg.org/#concept-request-destination)
- `event.request.destination === "unknown"` for direct access to `https://../script.js`
  - This is NOT standardized.
- `event.request.url` has path names ending with `.js` for scripts

## Warning
- This change blocks access to `view-source:https://.../script.js` and `https://.../script.js` but does **NOT block "Save As..." sources** since it seems that Service Worker is **BYPASSED** for "Save As..."
- So this change is pending as it is almost meaningless.

```diff
diff --git a/lib/service-worker.js b/lib/service-worker.js
index 0439365..ef5bb53 100644
--- a/lib/service-worker.js
+++ b/lib/service-worker.js
@@ -49,6 +49,17 @@ module.exports = function (hook, preprocess) {
     }
     event.respondWith(
       caches.open('version_' + version).then(function(cache) {
+        if ((new URL(event.request.url)).pathname.match(/[.]m?js$/)) {
+          switch (event.request.destination) {
+          case 'document':
+          case 'unknown':
+            console.warn('blocking access to ' + event.request.url);
+            return new Response('', { headers: { 'Content-Type': 'text/plain' } });
+          case '':
+          default:
+            break;
+          }
+        }
         return (restarted
           ? cache.match(CONFIGURATIONS_PSEUDO_URL).then(function(configurationsResponse) {
             if (configurationsResponse) {
```

**Comment by t2ym** [comment]
## Possible workaround to block "Save As..." script sources
- If the source file is NOT directly fetched, `Save As...` of the current Chrome (version 65) cannot save the source file.

## Notes
- Generation of `exported.json`, which contains the archived caches of the hooked scripts, is not automated yet.  It has to be manually copied and pasted from the browser.

## Steps to verify this workaround in the demo
- Adjust the cache version as below to load the old `demo/exported.json` as if they contain the latest versions of hooked scripts`
```diff
diff --git a/demo/exported.html b/demo/exported.html
index effddce..a959050 100644
--- a/demo/exported.html
+++ b/demo/exported.html
@@ -3,7 +3,7 @@
     let exportedResponse = await fetch('exported.json');
     let exportedText = await exportedResponse.text();
     let exported = JSON.parse(exportedText);
-    let cache = await caches.open('version_478');
+    let cache = await caches.open('version_491');
     for (let key in exported) {
       let url = new URL(key, location);
       let extension = url.pathname.match(/([.][a-z]*)$/)[1];
```
- Navigate to `/components/thin-hook/demo/exported.html`
  - Wait for the loading completion (Note: some scripts are old versions)
- Navigate to `/components/thin-hook/demo/normalize.js`
  - Hooked script should be shown with `hook.min.js 0.0.228`
- Try to download the source file with `Save As...`
- Check the downloaded source file
  - **The downloaded source file does NOT exist**

**Comment by t2ym** [comment]
Fixed via #252
